content/graphql/resolvers-map.md

```diff
@@ -28,11 +41,13 @@ export class Author {
 }
 ```
 
-> info **Hint** TypeScript's metadata reflection system has several limitations which make it impossible to, for instance, determine what properties a class consists of or recognize whether a given property is optional or required. Thus, we must either use the `@Field` decorator or use a [CLI plugin](/graphql/resolvers#cli-plugin).
```

Comment:
Side topic - do you think we should add `@filename()` directives to code snippets? For instance, for `Author` we could use `author.entity` (or model) etc

Reply:
Yes, I definitely do!  I think it would help a **lot** to provide some guidance on how to organize code in files.  Would you be able to do that (I'm not 100% sure I can make the best recommendations)!

Reply:
Perfect 💪 Example structure:

```
// AUTHOR MODULE (`authors` directory)
- dto
   - get-author.args.ts
- models
   - author.model.ts
authors.module.ts
authors.service.ts
authors.resolver.ts

// POST MODULE (`posts` directory)
- dto
   - create-post.input.ts
- models
   - post.model.ts
posts.module.ts
posts.service.ts
```

Conventionally:

- unions: `NAME.union.ts`
- enums: `NAME.enum.ts`
- plugins: `NAME.plugin.ts`
- directives: `NAME.directive.ts`

And as for interfaces, I think we should use `.model.ts` suffix as well (to avoid confusion with TS interfaces).

Reply:
Done for resolvers.  Will have to refer back to this as I go through other chapters.